### ISSUE TITLE
OCI_GetString returns null on implicit conversion of a RAW value when it's longer than 64 bytes

### ISSUE THREAD

**Issue by iliasaz** [vrogier/ocilib#288]
Hello,

I'm using OCI_GetString to implicitly convert various data types to a string, and it returns null for RAW values longer than 64 bytes. Here is a test case.

DB setup:

```
drop table rawtest;
create table rawtest (id number, val raw(2000));
insert into rawtest
select lvl-1, hextoraw(val) from (
  with l as (select level as lvl from dual connect by level < 1002)
  select lvl, listagg('FF','') within group (order by null) as val 
  from l connect by level < lvl and prior lvl = lvl and prior sys_guid() is not null 
  group by lvl
) where lvl > 1;
commit;
select * from rawtest order by id;
```

C code:

```
#include "ocilib.h"

int main(int argc, char *argv[])
{
    OCI_Connection* cn;
    OCI_Statement* st;
    OCI_Resultset* rs;
    OCI_Initialize(NULL, NULL, OCI_ENV_DEFAULT);
 
    cn = OCI_ConnectionCreate("database", "user", "password", OCI_SESSION_DEFAULT);
    st = OCI_StatementCreate(cn);
 
    OCI_ExecuteStmt(st, "select * from rawtest");
 
    rs = OCI_GetResultset(st);
 
    while (OCI_FetchNext(rs))
    {
        printf("%i - %s\n", OCI_GetInt(rs, 1), OCI_GetString(rs,2));
    }
 
    OCI_StatementFree(st);
    OCI_ConnectionFree(cn);
    OCI_Cleanup();
 
    return EXIT_SUCCESS;
}
```

Results:

```
1 - FF
2 - FFFF
3 - FFFFFF
.........
62 - FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF
63 - FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF
64 - FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF
65 - (null)
66 - (null)
67 - (null)
68 - (null)
```

My guess is that the buffer size is calculated incorrectly somewhere in OcilibResultsetGetString. Could you please check when you get a chance?

Thanks!

**Comment by vrogier** [comment]
Hi,

Thanks for reporting the issue :)

I pushed a fix and non regression test for it on v4.7.4 branch.

Regards,

Vincent

**Comment by iliasaz** [comment]
Hi Vincent,

Works great 👍  Thank you for fixing it so quickly.

Ilia
